### DISCUSSION TITLE
Better management of partially paid invoices via API

### DISCUSSION THREAD

**Discussion by shawn-digitalpoint** [btcpayserver/btcpayserver#2571] [Technical Support]
It would be nice to have an automatic way to handle partially paid invoices.  Say a user owed $10,000 (invoice created for that amount), but then paid $5,000 before the invoice expired.  In our case, it would be nice if we could put that amount as a credit on the user's account (handled on our end of course, not a btcpay server thing).

What I propose would be the addition of a new webhook for "invoice partially settled" that would fire when all payments for the invoice that started before the invoice expired have confirmed (as long as confirmations happened within the normal confirmation window times).  

Example:
- Settings: invoice expires after 60 minutes and payments must be confirmed within 1440 minutes.

1. Invoice is created for $10,000.
2. User pays $5,000 after 10 minutes, $1,000 more after 20 minutes.

If both payments confirmed before invoice expiration time, fire the webhook when invoice expires.
If both payments confirm an hour after invoice expired, fire the webhook at that point.

It would be nice if the webhook broke down each payment made to the invoice (basically the same thing you get from the invoice payment methods API call).  Then we could handle crediting a user on our side with payments made in whatever currency and they could use that credit in the future.

**Comment by nosovk** [comment]
It seems that I created duplicated question here https://github.com/btcpayserver/btcpayserver/discussions/2763
Were you able to find a solution for yours case?
